[WINDOWS] Most LSPs give error: "Cannot read LSP message header"

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Don't know, I just did compile to Windows and all suddenly most LSPs gave this error.

### Environment

~ via  v20.11.0
❯ Get-ComputerInfo


WindowsBuildLabEx                                       : 22621.1.amd64fre.ni_release.220506-1250
WindowsCurrentVersion                                   : 6.3
WindowsEditionId                                        : Core
WindowsInstallationType                                 : Client
WindowsInstallDateFromRegistry                          : 24/01/2024 11:10:49
WindowsProductId                                        : 00325-96698-25104-AAOEM
WindowsProductName                                      : Windows 10 Home
WindowsRegisteredOrganization                           :
WindowsRegisteredOwner                                  : efdal.pix4@outlook.com
WindowsSystemRoot                                       : C:\WINDOWS
WindowsVersion                                          : 2009
OSDisplayVersion                                        : 23H2
BiosCharacteristics                                     : {7, 11, 12, 15...}
BiosBIOSVersion                                         : {_ASUS_ - 1072009, FA506IV.320, American Megatrends - 50010}
BiosBuildNumber                                         :
BiosCaption                                             : FA506IV.320
BiosCodeSet                                             :
BiosCurrentLanguage                                     : en|US|iso8859-1
BiosDescription                                         : FA506IV.320
BiosEmbeddedControllerMajorVersion                      : 3
BiosEmbeddedControllerMinorVersion                      : 15
BiosFirmwareType                                        : Uefi
BiosIdentificationCode                                  :
BiosInstallableLanguages                                : 1
BiosInstallDate                                         :
BiosLanguageEdition                                     :
BiosListOfLanguages                                     : {en|US|iso8859-1}
BiosManufacturer                                        : American Megatrends Inc.
BiosName                                                : FA506IV.320
BiosOtherTargetOS                                       :
BiosPrimaryBIOS                                         : True
BiosReleaseDate                                         : 01/06/2022 02:00:00
BiosSeralNumber                                         : L8NRCX01B611329
BiosSMBIOSBIOSVersion                                   : FA506IV.320
BiosSMBIOSMajorVersion                                  : 3
BiosSMBIOSMinorVersion                                  : 2
BiosSMBIOSPresent                                       : True
BiosSoftwareElementState                                : Running
BiosStatus                                              : OK
BiosSystemBiosMajorVersion                              : 5
BiosSystemBiosMinorVersion                              : 16
BiosTargetOperatingSystem                               : 0
BiosVersion                                             : _ASUS_ - 1072009
CsAdminPasswordStatus                                   : Unknown
CsAutomaticManagedPagefile                              : True
CsAutomaticResetBootOption                              : True
CsAutomaticResetCapability                              : True
CsBootOptionOnLimit                                     :
CsBootOptionOnWatchDog                                  :
CsBootROMSupported                                      : True
CsBootStatus                                            : {0, 0, 0, 0...}
CsBootupState                                           : Normal boot
CsCaption                                               : DESKTOP-0D8QCTN
CsChassisBootupState                                    : Safe
CsChassisSKUNumber                                      : Default string
CsCurrentTimeZone                                       : 120
CsDaylightInEffect                                      : True
CsDescription                                           : AT/AT COMPATIBLE
CsDNSHostName                                           : DESKTOP-0D8QCTN
CsDomain                                                : WORKGROUP
CsDomainRole                                            : StandaloneWorkstation
CsEnableDaylightSavingsTime                             : True
CsFrontPanelResetStatus                                 : Unknown
CsHypervisorPresent                                     : False
CsInfraredSupported                                     : False
CsInitialLoadInfo                                       :
CsInstallDate                                           :
CsKeyboardPasswordStatus                                : Unknown
CsLastLoadInfo                                          :
CsManufacturer                                          : ASUSTeK COMPUTER INC.
CsModel                                                 : ASUS TUF Gaming A15 FA506IV_FX506IV
CsName                                                  : DESKTOP-0D8QCTN
CsNetworkAdapters                                       : {WiFi, Bluetooth Network Connection, Ethernet}
CsNetworkServerModeEnabled                              : True
CsNumberOfLogicalProcessors                             : 16
CsNumberOfProcessors                                    : 1
CsProcessors                                            : {AMD Ryzen 7 4800H with Radeon Graphics         }
CsOEMStringArray                                        : {  ,  ,  ,  ...}
CsPartOfDomain                                          : False
CsPauseAfterReset                                       : -1
CsPCSystemType                                          : Mobile
CsPCSystemTypeEx                                        : Mobile
CsPowerManagementCapabilities                           :
CsPowerManagementSupported                              :
CsPowerOnPasswordStatus                                 : Unknown
CsPowerState                                            : Unknown
CsPowerSupplyState                                      : Safe
CsPrimaryOwnerContact                                   :
CsResetCapability                                       : Other
CsResetCount                                            : -1
CsResetLimit                                            : -1
CsRoles                                                 : {LM_Workstation, LM_Server, NT}
CsStatus                                                : OK
CsSupportContactDescription                             :
CsSystemFamily                                          : ASUS TUF Gaming A15
CsSystemSKUNumber                                       :
CsSystemType                                            : x64-based PC
CsThermalState                                          : Safe
CsTotalPhysicalMemory                                   : 16556679168
CsPhyicallyInstalledMemory                              : 16777216
CsUserName                                              : DESKTOP-0D8QCTN\efdal
CsWakeUpType                                            : PowerSwitch
CsWorkgroup                                             : WORKGROUP
OsName                                                  : Microsoft Windows 11 Home
OsType                                                  : WINNT
OsOperatingSystemSKU                                    : WindowsHome
OsVersion                                               : 10.0.22631
OsCSDVersion                                            :
OsBuildNumber                                           : 22631
OsHotFixes                                              : {KB5037591, KB5027397, KB5037771, KB5037020...}
OsBootDevice                                            : \Device\HarddiskVolume1
OsSystemDevice                                          : \Device\HarddiskVolume3
OsSystemDirectory                                       : C:\WINDOWS\system32
OsSystemDrive                                           : C:
OsWindowsDirectory                                      : C:\WINDOWS
OsCountryCode                                           : 44
OsCurrentTimeZone                                       : 120
OsLocaleID                                              : 0809
OsLocale                                                : en-GB
OsLocalDateTime                                         : 18/05/2024 09:25:43
OsLastBootUpTime                                        : 17/05/2024 13:26:54
OsUptime                                                : 19:58:48.1187349
OsBuildType                                             : Multiprocessor Free
OsCodeSet                                               : 1252
OsDataExecutionPreventionAvailable                      : True
OsDataExecutionPrevention32BitApplications              : True
OsDataExecutionPreventionDrivers                        : True
OsDataExecutionPreventionSupportPolicy                  : OptIn
OsDebug                                                 : False
OsDistributed                                           : False
OsEncryptionLevel                                       : 256
OsForegroundApplicationBoost                            : Maximum
OsTotalVisibleMemorySize                                : 16168632
OsFreePhysicalMemory                                    : 6381824
OsTotalVirtualMemorySize                                : 21673656
OsFreeVirtualMemory                                     : 8281472
OsInUseVirtualMemory                                    : 13392184
OsTotalSwapSpaceSize                                    :
OsSizeStoredInPagingFiles                               : 5505024
OsFreeSpaceInPagingFiles                                : 5351476
OsPagingFiles                                           : {C:\pagefile.sys}
OsHardwareAbstractionLayer                              : 10.0.22621.2506
OsInstallDate                                           : 24/01/2024 12:10:49
OsManufacturer                                          : Microsoft Corporation
OsMaxNumberOfProcesses                                  : 4294967295
OsMaxProcessMemorySize                                  : 137438953344
OsMuiLanguages                                          : {en-GB, en-US, nl-NL}
OsNumberOfLicensedUsers                                 :
OsNumberOfProcesses                                     : 269
OsNumberOfUsers                                         : 2
OsOrganization                                          :
OsArchitecture                                          : 64-bit
OsLanguage                                              : en-GB
OsProductSuites                                         : {TerminalServicesSingleSession, HomeEdition}
OsOtherTypeDescription                                  :
OsPAEEnabled                                            :
OsPortableOperatingSystem                               : False
OsPrimary                                               : True
OsProductType                                           : WorkStation
OsServicePackMajorVersion                               : 0
OsServicePackMinorVersion                               : 0
OsStatus                                                : OK
OsSuites                                                : {TerminalServices, TerminalServicesSingleSession,
                                                          HomeEdition}
OsServerLevel                                           :
KeyboardLayout                                          : 00020409
TimeZone                                                : (UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
LogonServer                                             : \\DESKTOP-0D8QCTN
PowerPlatformRole                                       : Mobile
HyperVisorPresent                                       : False
HyperVRequirementDataExecutionPreventionAvailable       : True
HyperVRequirementSecondLevelAddressTranslation          : True
HyperVRequirementVirtualizationFirmwareEnabled          : True
HyperVRequirementVMMonitorModeExtensions                : True
DeviceGuardSmartStatus                                  : Off
DeviceGuardRequiredSecurityProperties                   :
DeviceGuardAvailableSecurityProperties                  :
DeviceGuardSecurityServicesConfigured                   :
DeviceGuardSecurityServicesRunning                      :
DeviceGuardCodeIntegrityPolicyEnforcementStatus         : EnforcementMode
DeviceGuardUserModeCodeIntegrityPolicyEnforcementStatus : Off

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

[2024-05-18T09:19:47+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:47+02:00 ERROR project] failed to start language server "intelephense": oneshot canceled
[2024-05-18T09:19:47+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\php\\node_modules\\intelephense\\lib\\intelephense.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:47+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:48+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:48+02:00 ERROR project] failed to start language server "vscode-html-language-server": oneshot canceled
[2024-05-18T09:19:48+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\html\\node_modules\\vscode-langservers-extracted\\bin\\vscode-html-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:48+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:48+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:49+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:50+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:50+02:00 ERROR project] failed to start language server "eslint": oneshot canceled
[2024-05-18T09:19:50+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:50+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:54+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:54+02:00 ERROR project] failed to start language server "tailwindcss-language-server": oneshot canceled
[2024-05-18T09:19:54+02:00 ERROR project] server stderr: Some("C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules\\.bin\\tailwindcss-language-server:2\r\nbasedir=$(dirname \"$(echo \"$0\" | sed -e 's,\\\\,/,g')\")\r\n          ^^^^^^^\r\n\r\nSyntaxError: missing ) after argument list\r\n    at internalCompileFunction (node:internal/vm:73:18)\r\n    at wrapSafe (node:internal/modules/cjs/loader:1176:20)\r\n    at Module._compile (node:internal/modules/cjs/loader:1218:27)\r\n    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)\r\n    at Module.load (node:internal/modules/cjs/loader:1117:32)\r\n    at Module._load (node:internal/modules/cjs/loader:958:12)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:54+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:55+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:55+02:00 ERROR project] failed to start language server "intelephense": oneshot canceled
[2024-05-18T09:19:55+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\php\\node_modules\\intelephense\\lib\\intelephense.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:55+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:55+02:00 ERROR util] crates\languages\src\tailwind.rs:229: missing executable in directory "C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules"
[2024-05-18T09:19:55+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:55+02:00 ERROR project] failed to start language server "tailwindcss-language-server": oneshot canceled
[2024-05-18T09:19:55+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules\\.bin\\tailwindcss-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:55+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:56+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:56+02:00 ERROR project] failed to start language server "vscode-html-language-server": oneshot canceled
[2024-05-18T09:19:56+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\html\\node_modules\\vscode-langservers-extracted\\bin\\vscode-html-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:56+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:56+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:56+02:00 ERROR util] crates\languages\src\tailwind.rs:229: The system cannot find the path specified. (os error 3)
[2024-05-18T09:19:56+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:56+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:56+02:00 ERROR project] failed to start language server "tailwindcss-language-server": oneshot canceled
[2024-05-18T09:19:56+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules\\.bin\\tailwindcss-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:56+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:57+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:57+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:57+02:00 ERROR project] failed to start language server "intelephense": oneshot canceled
[2024-05-18T09:19:57+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\php\\node_modules\\intelephense\\lib\\intelephense.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:57+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:57+02:00 ERROR util] crates\languages\src\tailwind.rs:229: The system cannot find the path specified. (os error 3)
[2024-05-18T09:19:57+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:57+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:57+02:00 ERROR project] failed to start language server "tailwindcss-language-server": oneshot canceled
[2024-05-18T09:19:57+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules\\.bin\\tailwindcss-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:57+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:58+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:58+02:00 ERROR project] failed to start language server "eslint": oneshot canceled
[2024-05-18T09:19:58+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:58+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:58+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:58+02:00 ERROR project] failed to start language server "vscode-html-language-server": oneshot canceled
[2024-05-18T09:19:58+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n")
[2024-05-18T09:19:58+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:58+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:58+02:00 ERROR util] crates\languages\src\tailwind.rs:229: The system cannot find the path specified. (os error 3)
[2024-05-18T09:19:58+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:59+02:00 ERROR project] failed to start language server "tailwindcss-language-server": oneshot canceled
[2024-05-18T09:19:59+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules\\.bin\\tailwindcss-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:59+02:00 ERROR project] Hit 4 reinstallation attempts for "tailwindcss-language-server"
[2024-05-18T09:19:59+02:00 ERROR util] crates\languages\src\typescript.rs:433: The system cannot find the path specified. (os error 3)
[2024-05-18T09:19:59+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:59+02:00 ERROR project] failed to start language server "eslint": oneshot canceled
[2024-05-18T09:19:59+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:19:59+02:00 ERROR project] failed to start language server "intelephense": oneshot canceled
[2024-05-18T09:19:59+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\php\\node_modules\\intelephense\\lib\\intelephense.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:19:59+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:19:59+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:20:00+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:20:00+02:00 ERROR project] failed to start language server "vscode-html-language-server": oneshot canceled
[2024-05-18T09:20:00+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\html\\node_modules\\vscode-langservers-extracted\\bin\\vscode-html-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:20:00+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:20:00+02:00 ERROR util] crates\languages\src\typescript.rs:433: The system cannot find the path specified. (os error 3)
[2024-05-18T09:20:00+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:20:00+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:20:00+02:00 ERROR project] failed to start language server "eslint": oneshot canceled
[2024-05-18T09:20:00+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:20:00+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:20:00+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:20:01+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:20:01+02:00 ERROR project] failed to start language server "intelephense": oneshot canceled
[2024-05-18T09:20:01+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\php\\node_modules\\intelephense\\lib\\intelephense.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:20:01+02:00 ERROR project] Hit 4 reinstallation attempts for "intelephense"
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:20:01+02:00 ERROR util] crates\languages\src\typescript.rs:433: The system cannot find the path specified. (os error 3)
[2024-05-18T09:20:01+02:00 ERROR util] crates\language\src\language_registry.rs:877: server container removal

Caused by:
    The system cannot find the file specified. (os error 2)
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:20:01+02:00 ERROR project] failed to start language server "eslint": oneshot canceled
[2024-05-18T09:20:01+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\Users\\efdal\\AppData\\Local\\Zed\\languages\\eslint\\vscode-eslint-release\\vscode-eslint\\server\\out\\eslintServer.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:20:01+02:00 ERROR project] Hit 4 reinstallation attempts for "eslint"
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:357: cannot read LSP message headers
[2024-05-18T09:20:01+02:00 ERROR project] failed to start language server "vscode-html-language-server": oneshot canceled
[2024-05-18T09:20:01+02:00 ERROR project] server stderr: Some("node:internal/modules/cjs/loader:1078\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'C:\\C:\\Users\\efdal\\AppData\\Local\\Zed\\extensions\\work\\html\\node_modules\\vscode-langservers-extracted\\bin\\vscode-html-language-server'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)\r\n    at Module._load (node:internal/modules/cjs/loader:920:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.15.0\r\n")
[2024-05-18T09:20:01+02:00 ERROR util] crates\lsp\src/lsp.rs:773: server shut down
[2024-05-18T09:20:01+02:00 ERROR project] Hit 4 reinstallation attempts for "vscode-html-language-server"

Hi, I was having the same error with several LSPs. For example, when Zed tried to start a new LSP for Svelte, it threw the following error:
```
2024-10-15T21:08:45.5659771+02:00 [INFO] starting language server process. binary path: "C:\\Program Files\\nodejs\\node.exe", working directory: "I:\\ProyectosWeb\\GrowFund", args: ["/C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte/node_modules/svelte-language-server/bin/server.js", "--stdio"]
2024-10-15T21:08:45.6236236+02:00 [ERROR] cannot read LSP message headers
2024-10-15T21:08:45.6239995+02:00 [ERROR] Failed to start language server "svelte-language-server": oneshot canceled
2024-10-15T21:08:45.6241354+02:00 [ERROR] server stderr: "node:internal/modules/cjs/loader:1080\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'I:\\C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte\\node_modules\\svelte-language-server\\bin\\server.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)\r\n    at Module._load (node:internal/modules/cjs/loader:922:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.18.0\r\n"
```

The problem is that in the args, Zed somehow adds a slash at the beginning, letting Windows take that slash as the root of the working directory, leaving the result as I:\C:\ in my case.

The workaround until a solid one arrives, is as follows:
Add a replace in the arg variable to change the slash in **crates/extension/src/extension_lsp_adapter.rs** line **91**.

Old:
```
Ok(LanguageServerBinary {
    path,
    arguments: command.args.into_iter().map(|arg| arg.into()).collect(),
    env: Some(command.env.into_iter().collect()),
})
```

New:
```
Ok(LanguageServerBinary {
    path,
    arguments: command.args.into_iter().map(|arg| arg.replace("/C:\\", "C:\\").into()).collect(),
    env: Some(command.env.into_iter().collect()),
})
```

Now, the log prints the following:
```
2024-10-15T23:31:25.6712345+02:00 [INFO] starting language server process. binary path: "C:\\Program Files\\nodejs\\node.exe", working directory: "I:\\ProyectosWeb\\GrowFund", args: ["C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte/node_modules/svelte-language-server/bin/server.js", "--stdio"]
```
As a result, Svelte suggestions/imports and Emmet inside .svelte files works perfectly.

I'm posting this comment because it worked for me and I'll be glad if it helps anyone else. I know this is a crappy solution, but at least, the extensions are working on Windows.

If anyone has another solution, I would be happy to see it.

@JoelDoryoku 
```
Language server error: svelte-language-server

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1249
  throw err;
  ^

Error: Cannot find module 'C:\C:\Users\v\AppData\Local\Zed\extensions\work\svelte\node_modules\svelte-language-server\bin\server.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1246:15)
    at Function._load (node:internal/modules/cjs/loader:1072:27)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:216:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v23.1.0
```

> Hi, I was having the same error with several LSPs. For example, when Zed tried to start a new LSP for Svelte, it threw the following error:
> 
> ```
> 2024-10-15T21:08:45.5659771+02:00 [INFO] starting language server process. binary path: "C:\\Program Files\\nodejs\\node.exe", working directory: "I:\\ProyectosWeb\\GrowFund", args: ["/C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte/node_modules/svelte-language-server/bin/server.js", "--stdio"]
> 2024-10-15T21:08:45.6236236+02:00 [ERROR] cannot read LSP message headers
> 2024-10-15T21:08:45.6239995+02:00 [ERROR] Failed to start language server "svelte-language-server": oneshot canceled
> 2024-10-15T21:08:45.6241354+02:00 [ERROR] server stderr: "node:internal/modules/cjs/loader:1080\r\n  throw err;\r\n  ^\r\n\r\nError: Cannot find module 'I:\\C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte\\node_modules\\svelte-language-server\\bin\\server.js'\r\n    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)\r\n    at Module._load (node:internal/modules/cjs/loader:922:27)\r\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12)\r\n    at node:internal/main/run_main_module:23:47 {\r\n  code: 'MODULE_NOT_FOUND',\r\n  requireStack: []\r\n}\r\n\r\nNode.js v18.18.0\r\n"
> ```
> 
> The problem is that in the args, Zed somehow adds a slash at the beginning, letting Windows take that slash as the root of the working directory, leaving the result as I:\C:\ in my case.
> 
> The workaround until a solid one arrives, is as follows: Add a replace in the arg variable to change the slash in **crates/extension/src/extension_lsp_adapter.rs** line **91**.
> 
> Old:
> 
> ```
> Ok(LanguageServerBinary {
>     path,
>     arguments: command.args.into_iter().map(|arg| arg.into()).collect(),
>     env: Some(command.env.into_iter().collect()),
> })
> ```
> 
> New:
> 
> ```
> Ok(LanguageServerBinary {
>     path,
>     arguments: command.args.into_iter().map(|arg| arg.replace("/C:\\", "C:\\").into()).collect(),
>     env: Some(command.env.into_iter().collect()),
> })
> ```
> 
> Now, the log prints the following:
> 
> ```
> 2024-10-15T23:31:25.6712345+02:00 [INFO] starting language server process. binary path: "C:\\Program Files\\nodejs\\node.exe", working directory: "I:\\ProyectosWeb\\GrowFund", args: ["C:\\Users\\joell\\AppData\\Local\\Zed\\extensions\\work\\svelte/node_modules/svelte-language-server/bin/server.js", "--stdio"]
> ```
> 
> As a result, Svelte suggestions/imports and Emmet inside .svelte files works perfectly.
> 
> I'm posting this comment because it worked for me and I'll be glad if it helps anyone else. I know this is a crappy solution, but at least, the extensions are working on Windows.
> 
> If anyone has another solution, I would be happy to see it.

@lilhammer111 Try this, but now the fix is in crates/language_extensions/src/extension_lsp_adapter.rs line 158

--- a/crates/language_extension/src/extension_lsp_adapter.rs
+++ b/crates/language_extension/src/extension_lsp_adapter.rs
@@ -1,5 +1,5 @@
 use std::ops::Range;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
 use anyhow::{Context as _, Result};
@@ -174,7 +174,32 @@ impl DynLspInstaller for ExtensionLspAdapter {
                     )
                     .await?;
 
-                let path = self.extension.path_from_extension(command.command.as_ref());
+                // on windows, extensions might produce weird paths
+                // that start with a leading slash due to WASI
+                // requiring that for PWD and friends so account for
+                // that here and try to transform those paths back
+                // to windows paths
+                //
+                // if we don't do this, std will interpret the path as relative,
+                // which changes join behavior
+                let command_path: &Path = if cfg!(windows)
+                    && let Some(command) = command.command.to_str()
+                {
+                    let mut chars = command.chars();
+                    if chars.next().is_some_and(|c| c == '/')
+                        && chars.next().is_some_and(|c| c.is_ascii_alphabetic())
+                        && chars.next().is_some_and(|c| c == ':')
+                        && chars.next().is_some_and(|c| c == '\\' || c == '/')
+                    {
+                        // looks like a windows path with a leading slash, so strip it
+                        command.strip_prefix('/').unwrap().as_ref()
+                    } else {
+                        command.as_ref()
+                    }
+                } else {
+                    command.command.as_ref()
+                };
+                let path = self.extension.path_from_extension(command_path);
 
                 // TODO: This should now be done via the `zed::make_file_executable` function in
                 // Zed extension API, but we're leaving these existing usages in place temporarily
@@ -193,7 +218,32 @@ impl DynLspInstaller for ExtensionLspAdapter {
 
                 Ok(LanguageServerBinary {
                     path,
-                    arguments: command.args.into_iter().map(|arg| arg.into()).collect(),
+                    arguments: command
+                        .args
+                        .into_iter()
+                        .map(|arg| {
+                            // on windows, extensions might produce weird paths
+                            // that start with a leading slash due to WASI
+                            // requiring that for PWD and friends so account for
+                            // that here and try to transform those paths back
+                            // to windows paths
+                            if cfg!(windows) {
+                                let mut chars = arg.chars();
+                                if chars.next().is_some_and(|c| c == '/')
+                                    && chars.next().is_some_and(|c| c.is_ascii_alphabetic())
+                                    && chars.next().is_some_and(|c| c == ':')
+                                    && chars.next().is_some_and(|c| c == '\\' || c == '/')
+                                {
+                                    // looks like a windows path with a leading slash, so strip it
+                                    arg.strip_prefix('/').unwrap().into()
+                                } else {
+                                    arg.into()
+                                }
+                            } else {
+                                arg.into()
+                            }
+                        })
+                        .collect(),
                     env: Some(command.env.into_iter().collect()),
                 })
             })

